include/seqan3/alphabet/quality/phred42.hpp

```diff
@@ -42,129 +42,151 @@ namespace seqan3
     using rank_type = uint8_t;
-    //! the char representation type of a quality score
+
+    //!\brief The integer representation of a quality score.
+    //!\hideinitializer
+    using phred_type = uint8_t;
+
+    //!\brief The '!'-based character representation of a quality score.
+    //!\hideinitializer
     using char_type = char;
+    //!\}
 
-    //! internal rank value representation
-    rank_type value;
+    /*!\name Member variables.
+    * \{
+    */
+    //!\brief The internal 0-based rank value.
+    rank_type _value;
 
-    //! projection offset of a char quality score
+    //!\brief The projection offset between char and rank quality score representation.
+    //!\hideinitializer
     static constexpr char_type offset_char{'!'};
-    //! projection offsets of a phred quality score
-    static constexpr phred_type offset_phred{0};
+    //!\}
 
-    //! implicit compatibility to inner_type
-    constexpr illumina18 & operator =(rank_type const c)
+    //!\brief Value assignment with implicit compatibility to inner type.
+    constexpr phred42 & operator =(phred_type const c)
     {
-        value = c;
+        assert(c < max_value_size);
+        _value = (c < value_size) ? c : value_size - 1;
         return *this;
     }
 
-    //!\name Comparison operators
-    //!\{
-    constexpr bool operator==(const illumina18 & rhs) const
+    /*!\name Comparison operators.
+    * \{
+    */
+    constexpr bool operator==(const phred42 & rhs) const
     {
-        return this->value == rhs.value;
+        return this->_value == rhs._value;
     }
 
-    constexpr bool operator!=(const illumina18 & rhs) const
+    constexpr bool operator!=(const phred42 & rhs) const
     {
-        return this->value != rhs.value;
+        return this->_value != rhs._value;
     }
 
-    constexpr bool operator<(const illumina18 & rhs) const
+    constexpr bool operator<(const phred42 & rhs) const
     {
-        return this->value < rhs.value;
+        return this->_value < rhs._value;
     }
 
-    constexpr bool operator>(const illumina18 & rhs) const
+    constexpr bool operator>(const phred42 & rhs) const
     {
-        return this->value > rhs.value;
+        return this->_value > rhs._value;
     }
 
-    constexpr bool operator<=(const illumina18 & rhs) const
+    constexpr bool operator<=(const phred42 & rhs) const
     {
-        return this->value <= rhs.value;
+        return this->_value <= rhs._value;
     }
 
-    constexpr bool operator>=(const illumina18 & rhs) const
+    constexpr bool operator>=(const phred42 & rhs) const
     {
-        return this->value >= rhs.value;
+        return this->_value >= rhs._value;
     }
     //!\}
 
-    //! explicit compatibility to char code of a quality score
+    /*!\name Conversion and explicit conversion functions.
+    * \{
+   */
+    //!\brief Explicit compatibility to char code of a quality score.
     explicit constexpr operator char() const
     {
         return to_char();
     }
```

Comment:
please remove this as we don't have conversions to char or rank type anymore, see also the documentation on alphabets!